tests-cuda/test_1809_array_cuda_jit.py

```diff
@@ -0,0 +1,98 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import cupy as cp  # noqa: F401
+import numpy as np  # noqa: F401
+import pytest  # noqa: F401
+
+import awkward as ak  # noqa: F401
+
+try:
+    import numba as nb  # noqa: E402
+    import numba.cuda as nb_cuda  # noqa: E402
+    from numba import config
+
+    ak.numba.register_and_check()
+
+except ImportError:
+    nb = nb_cuda = None
+
+numbatest = pytest.mark.skipif(
+    nb is None or nb_cuda is None, reason="requires the numba and numba.cuda packages"
+)
+
+threads_per_block = 128
+blocks_per_grid = 12
+
+config.CUDA_LOW_OCCUPANCY_WARNINGS = False
+config.CUDA_WARN_ON_IMPLICIT_COPY = False
+
+
+@nb_cuda.jit(extensions=[ak.numba.ArrayViewArgHandler()])
+def multiply(array, n, out):
+    tid = nb_cuda.grid(1)
+    out[tid] = array[tid] * n
+
+
+@nb_cuda.jit(extensions=[ak.numba.ArrayViewArgHandler()])
```

Comment:
@agoose77 - should I just define this as `array_view_arg_handler = ArrayViewArgHandler()` in `numba.py` and use it here as `ak,numba.array_view_arg_handler`?

Reply:
My attitude towards the `ak._connect.numba` vs `ak.numba` modules is that anything in `ak._connect.numba` can assume that Numba is available and compatible with Awkward, whilst `ak.numba` needs to check these things. Although we should expect Awkward to be registered by Numba in most cases, I think we still need to enforce that. Therefore, I think we want to be able to detect whether Numba has been registered in the process of obtaining an instance of `ArrayViewArgHandler`.

There are two main approaches that stand out to me:

1. A non-callable `cuda_extension` module variable 
	```python
	nb_cuda.jit(extensions=[ak.numba.cuda_extension])
	```
	If we don't have a callable `ArrayViewArgHandler`, then we would need to check that Awkward's Numba machinery is registered in `ArrayViewArgHandler.prepare_args`. 
2. A factory function in `ak.numba`:
	```python
	nb_cuda.jit(extensions=[ak.numba.create_cuda_extension()])
	```
    This function checks that Numba is registered, before returning the instance `ak._connect.numba.arrayview_cuda.ArrayViewArgHandler()`. I suppose this could be a singleton, I don't think it matters?


Although I assume that `ArrayViewArgHandler.prepare_args` is not called that often (i.e. it's only called for Python invocations), it would be better to check for registration as infrequently as possible. I am leaning towards option (2)!